Mounted assets docs (#18657)
cloudflare/cloudflare-docs: src/content/docs/workers/static-assets/routing.mdx

/not-found -> 200 /index.html
/folder/doesnt/exist -> 200 /index.html
```

> ## Mounting assets on a path

Reviewer: ```suggestion
## Serving assets from a custom path
```

:::note
This feature requires Wrangler v3.xx.xx or later.
:::

> You can mount assets on a path by creating a directory structure that mirrors the desired path.
> For example, to serve assets from `example.com/blog`, create a `blog` directory in your asset directory.

Reviewer: ```suggestion
Like with any other Worker, [you can configure a Worker with assets to run on a subpath of your domain](https://developers.cloudflare.com/workers/configuration/routing/routes/). Assets in such a Worker such be nested in a directory structure that mirrors the desired path.

For example, to serve assets from `example.com/blog/*`, create a `blog` directory in your asset directory.
```

This has also made me realize that we probably need to support some more `not_found_handling` options so that people can SPA 200 with `./blog/index.html`... For now, they'll just have to put it at the root.

<FileTree>
- dist
	- blog
		- index.html
		- posts
			- post1.html
			- post2.html
</FileTree>

With a Wrangler configuration file like so:
<WranglerConfig>
```toml

> name = "mounted-assets-example"

Reviewer: ```suggestion
name = "assets-on-a-path-example"
```

main = "src/index.js"
route = "example.com/blog/*"

[assets]
directory = "dist"
```
</WranglerConfig>

> In this example, requests to `example.com/blog` will serve the `index.html` file, and requests to `example.com/blog/posts/post1` will serve the `post1.html` file.

Reviewer: ```suggestion
In this example, requests to `example.com/blog/` will serve the `index.html` file, and requests to `example.com/blog/posts/post1` will serve the `post1.html` file.
```

> If you have a file outside the mounted path, it will not be served. For example, if you have a `home.html` file in the root of your asset directory, it will not be served when requesting `example.com/blog/home`.

Reviewer: ```suggestion
If you have a file outside the mounted path, it will not be served. For example, if you have a `home.html` file in the root of your asset directory, it will not be served when requesting `example.com/blog/home`. However, if needed, these files can still be manually fetched over [the binding](https://developers.cloudflare.com/workers/static-assets/binding/#binding).
```

/not-found -> 200 /index.html
/folder/doesnt/exist -> 200 /index.html
```

## Serving assets from a custom path

:::note
This feature requires Wrangler v3.xx.xx or later.
:::

> Like with any other Worker, [you can configure a Worker with assets to run on a subpath of your domain](https://developers.cloudflare.com/workers/configuration/routing/routes/). Assets in such a Worker such be nested in a directory structure that mirrors the desired path.

Reviewer: ```suggestion
Like with any other Worker, [you can configure a Worker with assets to run on a subpath of your domain](https://developers.cloudflare.com/workers/configuration/routing/routes/). Assets in such a Worker must be nested in a directory structure that mirrors the desired path.
```
